react/features/whiteboard/middleware.ts

```diff
@@ -68,28 +68,73 @@ MiddlewareRegistry.register((store: IStore) => (next: Function) => async (action
                 collabServerUrl,
                 collabDetails
             });
+            raiseWhiteboardNotification('Instantiated');
 
             return;
         }
 
         if (action.isOpen) {
             focusWhiteboard(store);
+            raiseWhiteboardNotification('Shown');
 
             return;
         }
 
         dispatch(participantLeft(WHITEBOARD_ID, conference, { fakeParticipant: FakeParticipant.Whiteboard }));
+        raiseWhiteboardNotification('Hidden');
+
         break;
     }
     case RESET_WHITEBOARD: {
         dispatch(participantLeft(WHITEBOARD_ID, conference, { fakeParticipant: FakeParticipant.Whiteboard }));
+        raiseWhiteboardNotification('Reset');
+
         break;
     }
     }
 
     return next(action);
 });
 
+/**
+ * Raises the whiteboard status notifications changes (if API is enabled).
+ *
+ * @param {string} event - The whiteboard event to be raised.
+ * @returns {Function}
+ */
+function raiseWhiteboardNotification(event: string) {
+    switch (event) {
```

Comment:
Instead of the switch, can't we have something like this?
```
if (typeof APP !== 'undefined') {
       APP.API.notifyWhiteboardStatusChanged(event);
}
```

Also can you please rename the parameter to `status` and extract an enum type from all the states?